pxtlib/emitter/cloud.ts

```diff
@@ -2,8 +2,11 @@
 namespace pxt.Cloud {
     import Util = pxtc.Util;
 
+    const baseEndpoint = "https://www.makecode.com";
+    //const baseEndpoint = "http://localhost:5500";
```

Comment:
Actually, this might not work with staging.pxt.io
I'll need to rethink this

Reply:
Does the client app have a per-environment config file?

Reply:
@eanders-MS , can you think of any reason why we wouldn't use "window.location.origin" here ?

Reply:
That should work, but might need to special case it for localhost to fixup the port number.

Reply:
I decided against this change since "baseEndpoint" is only used when running in node or localhost, otherwise we use site relative "/api" as the api root which is effectively like using window.location.origin